src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameService.cs

```diff
@@ -84,6 +91,41 @@ internal class InlineRenameService : IInlineRenameService
                 _asyncListener);
 
             return new InlineRenameSessionInfo(ActiveSession);
+
+            static bool IsReadOnlyOrCannotNavigateToSpan(IInlineRenameInfo renameInfo, Document document, CancellationToken cancellationToken)
+            {
+                // This is unpleasant, but we do everything synchronously.  That's because we end up
+                // needing to make calls on the UI thread to determine if the locations of the symbol
+                // are in readonly buffer sections or not.  If we go pure async we have the following
+                // problem:
+                //   1) if we call ConfigureAwait(false), then we might call into the text buffer on 
+                //      the wrong thread.
+                //   2) if we try to call those pieces of code on the UI thread, then we will deadlock
+                //      as our caller often is doing a 'Wait' on us, and our UI calling code won't run.
+                if (renameInfo is SymbolInlineRenameInfo symbolInlineRenameInfo)
```

Comment:
so i don't love this cast.  i'll have a suggestion on how we can avoid this below(i hope).